helpers/src/db/pagination.rs

```diff
@@ -0,0 +1,69 @@
+use diesel::pg::Pg;
+use diesel::prelude::*;
+use diesel::query_builder::*;
+use diesel::query_dsl::methods::LoadQuery;
+use diesel::sql_types::BigInt;
+use diesel::QueryId;
+
+const DEFAULT_PER_PAGE: i64 = 10;
+
+pub trait Paginate: Sized {
+    fn paginate(self, page: i64) -> Paginated<Self>;
+}
+
+impl<T> Paginate for T {
+    fn paginate(self, page: i64) -> Paginated<Self> {
+        Paginated {
+            query: self,
+            per_page: DEFAULT_PER_PAGE,
+            page,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, QueryId)]
+pub struct Paginated<T> {
+    query: T,
+    page: i64,
+    per_page: i64,
+}
+
+impl<T> Paginated<T> {
+    pub fn per_page(self, per_page: i64) -> Self {
+        Paginated { per_page, ..self }
```

Comment:
This somehow reads a bit weirdly..

What do you think about
* 1
    ```rust
    pub fn per_page(self, per_page: i64) -> Self {
            self.per_page = per_page;
            self
    }
    ```
* 2
    ```rust
    pub fn per_page(&mut self, per_page: i64) -> Self {
            self.per_page = per_page;
            self.clone()
    }
    ```
    * this would also enable to use it in-place and not only trailing

Reply:
1. You would also need to make the self parameter mutable. Would be possible, if you find this easier to understand.
2. The struct `Paginated` has no trait `clone`.

Reply:
1. true, needs to be mutable
2. clone could be easily derived

In the end it's your decision, but I find the current solution slightly hard to grasp (but maybe I am just not used to that pattern).